python/desc/sims/GCRCatSimInterface/InstanceCatalogWriter.py

```diff
@@ -26,8 +27,73 @@
     TwinklesCompoundInstanceCatalog_DC2 as twinklesDESCQACompoundObject, \
     sprinklerCompound_DC2 as sprinklerDESCQACompoundObject, \
     TwinklesCatalogZPoint_DC2 as DESCQACat_Twinkles
+from . import DC2PhosimCatalogSN, SNFileDBObject
+
+__all__ = ['InstanceCatalogWriter', 'make_instcat_header', 'get_obs_md',
+           'snphosimcat']
+
+
+# Global `numpy.dtype` instance to define the types  
+# in the csv files being read 
+SNDTYPESR1p1 = np.dtype([('snid_in', int),
+                         ('x0_in', float),
+                         ('t0_in', float),
+                         ('x1_in', float),
+                         ('c_in', float),
+                         ('z_in', float),
+                         ('snra_in', float),
+                         ('sndec_in', float)])
+
+def snphosimcat(fname, tableName, obs_metadata, objectIDtype, idColKey='snid_in',
+              delimiter=',', dtype=SNDTYPESR1p1):
+    """convenience function for writing out phosim instance catalogs for
+    different SN populations in DC2 Run 1.1 that have been serialized to
+    csv files.
+
+    Parameters:
+    -----------
+    fname : string
+        absolute path to csv file for SN population.
+    tableName : string 
+        table name describing the population to be decided by user choice.
+    obs_metadata: instance of `lsst.sims.utils.ObservationMetaData`
+	observation metadata describing the observation
+    idColKey : string, defaults to values for Run1.1
+        describes the input parameters to the database as interpreted from the
+        csv file.
+    delimiter : string, defaults to ','
+        delimiter used in the csv file
+    dtype : instance of `numpy.dtype`
+        tuples describing the variables and types in the csv files.
 
-__all__ = ['InstanceCatalogWriter', 'make_instcat_header', 'get_obs_md']
+
+    Returns
+    -------
+    returns an instance of a Phosim Instance catalogs with appropriate
+    parameters set for the objects in the file and the obs_metadata.
+    """
+    dbobj = SNFileDBObject(fname, runtable=tableName,
+                       idColKey=idColKey, dtype=dtype,
+                       delimiter=delimiter)
+    dbobj.raColName = 'snra_in'
+    dbobj.decColName ='sndec_in'
+    dbobj.objectTypeId = objectIDtype
+    cat = DC2PhosimCatalogSN(db_obj=dbobj, obs_metadata=obs_metadata)
+    cat.surveyStartDate = 0.
+    cat.maxz = 1.4 # increasing max redshift
+    cat.maxTimeSNVisible = 150.0 # increasing for high z SN  
+    cat.phoSimHeaderMap = DefaultPhoSimHeaderMap
+    cat.writeSedFile = True
+
+    # This means that the the spectra written by phosim will 
+    # go to `spectra_files/Dynamic/specFileSN_* 
+    # Note: you want DC2PhosimCatalogSN.sep to be part of this prefix
+    # string. 
+    # We can arrange for the phosim output to just read the string 
+    # without directories or something else
+
```

Comment:
So it appears that the SED files will be written to a subdirectory `spectra_files/Dynamic` relative to the current working directory, correct?  If that works ok for Tom's phosim pipeline, I guess that is fine.  Could we add a line here like
```
    os.makedirs('spectra_files/Dynamic', exist_ok=True)
```
?  Since the path is not selectable by the user, we might as well make sure the directory exists before writing to it.  This would be helpful since this part of the sprinkler is only executed after the non-sprinkled objects have been written to disk.  That may take a couple of tens of minutes depending on the fov value, and it would be frustrating to have the code crash after all that time simply because that directory didn't already exist.

Reply:
Done